js/main.js

```diff
@@ -18,52 +18,101 @@ function imageLink(event) {
   $image.setAttribute('src', event.target.value);
 }
 
-var $form = document.querySelector('form');
+var $form = document.querySelector('form[data-view="entry-form"');
 var $title = document.querySelector('input[name="user-title"]');
-var $notes = document.querySelector('textarea');
+var $notes = document.querySelector('textarea[name="notes"]');
 
 $form.addEventListener('submit', dataSet);
 
 function dataSet(event) {
   event.preventDefault();
-  var singleData = {
-    title: $title.value,
-    photoURL: $imageLink.value,
-    notes: $notes.value,
-    entryId: data.nextEntryId++
-  };
-  data.entries.unshift(singleData);
-  $image.setAttribute('src', 'images/placeholder-image-square.jpg');
-  event.target.reset();
-  $dataViewEntries.className = '';
-  $form.className = 'hidden';
-  $ul.prepend(createEntries(data.entries[0]));
-  entryText();
+  if (data.editing === null) {
+    var singleData = {
+      title: $title.value,
+      photoURL: $imageLink.value,
+      notes: $notes.value,
+      entryId: data.nextEntryId++
+    };
+    data.entries.unshift(singleData);
+    $image.setAttribute('src', 'images/placeholder-image-square.jpg');
+    event.target.reset();
+    $dataViewEntries.className = '';
+    $form.className = 'hidden';
+    $newEntryTitle.className = 'font28px';
+    $editEntryTitle.className = 'font28px hidden';
+    $ul.prepend(createEntries(data.entries[0]));
+  } else {
+    for (var i = 0; i < data.entries.length; i++) {
+      var spot = document.querySelectorAll('li');
+      if (Number(data.editing.entryId) === Number(data.entries[i].entryId)) {
+        data.editing.title = $title.value;
+        data.editing.photoURL = $imageLink.value;
+        data.editing.notes = $notes.value;
+        data.editing.entryId = data.entries[i].entryId;
+        data.entries[i].title = data.editing.title;
+        data.entries[i].photoURL = data.editing.photoURL;
+        data.entries[i].notes = data.editing.notes;
+        data.entries[i].entryId = data.editing.entryId;
```

Comment:
This can probably be deleted, based on what we discussed earlier about the nature of how objects are stored. `data.editing` **_is_** `data.entries[i]`